### PR TITLE
LPD-37404

### PR DESCRIPTION
**Pull request by bakayattila**
### What is this trying to solve?
https://liferay.atlassian.net/browse/LPD-37404

### How am I fixing it?
The issue is that we are not associated with the asset library in the current scope(the scop of the widget page template group). I used a similar solution to [TemplateSelectorTag.java#L321](https://github.com/liferay/liferay-portal/blob/master/modules/apps/template/template-taglib/src/main/java/com/liferay/template/taglib/servlet/taglib/TemplateSelectorTag.java#L321) This way I can retrieve the actual scope for LayoutPageTemplateEntry.

### How can you verify that it works?

To test run:
`gw testIntegration --tests AssetPublisherHelperTest.testGetItemSelectorScopeGroupWithLayoutPrototype`

Or manually test based on the LPD steps

### CODE REVIEW

**Comment by bakayattila** [comment]
ci:test:relevant

File: modules/apps/asset/asset-publisher-web/src/main/java/com/liferay/asset/publisher/web/internal/display/context/AssetPublisherDisplayContext.java [hunk 1]
```diff
@@ -1263,6 +1263,9 @@ public List<DropdownItem> getScopeDropdownItems(PortletURL addScopeURL)
 
 		dropdownItemList.add(
 			dropdownItem -> {
+				Group group = _assetPublisherHelper.getItemSelectorScopeGroup(
+					layout.getGroup());
+
```
Comment (adolfopa) [inline modules/apps/asset/asset-publisher-web/src/main/java/com/liferay/asset/publisher/web/internal/display/context/AssetPublisherDisplayContext.java:1268]: If this is used only here, why add the method to the API? Unless there's some other usages (or an established pattern for this), I think it is better to keep it in the display context as a private method.

**Review by adolfopa — CHANGES_REQUESTED** [review]

File: modules/apps/asset/asset-publisher-web/src/main/java/com/liferay/asset/publisher/web/internal/display/context/AssetPublisherDisplayContext.java (hunk 1 above)
Comment (bakayattila) [reply modules/apps/asset/asset-publisher-web/src/main/java/com/liferay/asset/publisher/web/internal/display/context/AssetPublisherDisplayContext.java:1268]: The use case is not really testable trough playwright, and as we use multiple services in the solution I would much rather prefer an integration test over a unit test.

Comment (adolfopa) [reply modules/apps/asset/asset-publisher-web/src/main/java/com/liferay/asset/publisher/web/internal/display/context/AssetPublisherDisplayContext.java:1268]: Nevermind. I see the interface is marked as provider type, so it is not a major change. As it simplifies testing, go ahead and send it.

**Comment by bakayattila** [comment]
ci:test:relevant

**Review by adolfopa — APPROVED** [review]

**Comment by adolfopa** [comment]
ci:forward

**Comment by bakayattila** [comment]
ci:forward

**Pull request closed without merging**